packages/ui-extensions-dev-console/src/sections/Extensions/components/QRCodeModal/QRCodeModal.tsx

```diff
@@ -1,94 +1,81 @@
 import * as styles from './QRCodeModal.module.scss'
 import React, {useCallback, useMemo} from 'react'
-import {CircleAlertMajor, DuplicateMinor} from '@shopify/polaris-icons'
-import {Button, Icon, Modal, ModalProps, Stack} from '@shopify/polaris'
+import {DuplicateMinor} from '@shopify/polaris-icons'
+import {Button, Modal, ModalProps} from '@shopify/polaris'
 import {useI18n} from '@shopify/react-i18n'
 import copyToClipboard from 'copy-to-clipboard'
 import QRCode from 'qrcode.react'
-import {ExtensionPayload} from '@shopify/ui-extensions-server-kit'
 import {toast} from 'react-toastify'
+import {ExtensionPayload} from '@shopify/ui-extensions-server-kit'
 import {useExtensionsInternal} from '@/sections/Extensions/hooks/useExtensionsInternal'
 
 export interface QRCodeModalProps extends Pick<ModalProps, 'open' | 'onClose'> {
-  extension?: ExtensionPayload
+  url?: string
+  title?: string
+  type?: ExtensionPayload['surface'] | 'home'
 }
 
-export function QRCodeModal({extension, open, onClose}: QRCodeModalProps) {
+export function QRCodeModal({url, title, type, ...modalProps}: QRCodeModalProps) {
   const [i18n] = useI18n({
     id: 'QRCodeModal',
     fallback: en,
   })
 
+  const content = url && title && type ? <QRCodeContent type={type} title={title} url={url} /> : null
+
   return (
-    <Modal title={i18n.translate('title')} titleHidden open={open} onClose={onClose} sectioned small noScroll>
-      <QRCodeContent extension={extension} />
+    <Modal title={i18n.translate('title', {title})} {...modalProps} titleHidden sectioned small noScroll>
+      {content}
     </Modal>
   )
 }
 
-export function QRCodeContent(props: Pick<QRCodeModalProps, 'extension'>) {
+export interface QRCodeContentProps {
+  url: string
+  title: string
+  type: ExtensionPayload['surface'] | 'home'
+}
+
+export function QRCodeContent({url, title, type}: QRCodeContentProps) {
   const [i18n] = useI18n({
     id: 'QRCodeModal',
     fallback: en,
   })
-  const {extension} = props
-  const {state} = useExtensionsInternal()
 
-  const mobileQRCode = useMemo(() => {
-    if (!state.app || !extension) {
-      return undefined
-    }
+  const {
+    state: {store, app},
+  } = useExtensionsInternal()
 
-    if (extension.surface === 'pos') {
-      return `com.shopify.pos://pos-ui-extensions?url=${extension.development.root.url}`
-    } else {
-      return `https://${state.store}/admin/extensions-dev/mobile?url=${extension.development.root.url}`
+  const apiKey = app?.apiKey
+  const appUrl = app?.url
+  const qrCodeURL = useMemo(() => {
+    if (type === 'pos') {
+      return `com.shopify.pos://pos-ui-extensions?url=${appUrl}`
     }
-  }, [extension, state.app, state.store])
 
-  const onButtonClick = useCallback(() => {
-    if (mobileQRCode && copyToClipboard(mobileQRCode)) {
-      toast(i18n.translate('qrcode.copied'), {toastId: `copy-qrcode-${mobileQRCode}`})
+    if (type === 'home') {
+      return `https://${store}/admin/apps/${apiKey}`
```

Comment:
Is this the URL we should hit on mobile? I tried following the QR code url for the app and I got to the app page with the ngrok "Visit Site" page displayed. Clicking "Visit Site" kept me on the same page. On desktop on the other hand I see the frontend index page after a few redirects and I see a few other params added like `shop` and `host`. Should we append those as well like `buildAppURL` does?

Reply:
Good catch, thank you!  This should be fixed now.